File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/CheckinCreateResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import java.time.LocalDateTime;

import com.ssafy.e106.domain.subscription.entity.CheckinRecord;

public record CheckinCreateResponse(
    Long checkinRecordId,
    Long serviceId,
    String cycleYm,
    String response,
    LocalDateTime respondedAt) {

  public static CheckinCreateResponse of(CheckinRecord checkinRecord) {
    return new CheckinCreateResponse(
        checkinRecord.getCheckinRecordId(),
        checkinRecord.getService().getServiceId(),
        checkinRecord.getCycleYm(),
        checkinRecord.getResponse().name(),
        checkinRecord.getRespondedAt());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/CheckInCreateResponse.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class CheckInCreateResponse(
    val checkinRecordId: Long,
    val serviceId: Long,
    val cycleYm: String,
    val response: String,
    val respondedAt: String,
)
